### PR TITLE
Support I/O error resiliency settings

### PR DESCRIPTION
**Pull request by derekbit**

### CODE REVIEW

File: vendor/github.com/longhorn/go-spdk-helper/pkg/spdk/client/basic.go [hunk 1]
```diff
@@ -432,15 +432,24 @@ func (c *Client) BdevRaidGetInfoByCategory(category spdktypes.BdevRaidCategory)
...
-			Trsvcid: trsvcid,
-			Adrfam:  adrfam,
+			Traddr:               traddr,
+			Trtype:               trtype,
+			Subnqn:               subnqn,
+			Trsvcid:              trsvcid,
+			Adrfam:               adrfam,
+			CtrlrLossTimeoutSec:  ctrlrLossTimeoutSec,
+			ReconnectDelaySec:    reconnectDelaySec,
+			FastIOFailTimeoutSec: reconnectDelaySec,
```
Comment (shuo-wu) [inline vendor/github.com/longhorn/go-spdk-helper/pkg/spdk/client/basic.go:452]: As mentioned in the original PR, typo here

File: proto/spdkrpc/spdk.proto [hunk 2]
```diff
@@ -98,6 +98,9 @@ message EngineCreateRequest {
     uint64 spec_size = 3;
     map<string, string> replica_address_map = 4;
     string frontend = 5;
+    int32 ctrlr_loss_timeout = 6;
+    int32 reconnect_delay = 7;
+    int32 fast_io_fail_timeout = 8;
```
Comment (shuo-wu) [inline proto/spdkrpc/spdk.proto:103]: Should we expose these in `Engine`?

File: vendor/github.com/longhorn/go-spdk-helper/pkg/spdk/client/basic.go (hunk 1 above)
Comment (derekbit) [reply vendor/github.com/longhorn/go-spdk-helper/pkg/spdk/client/basic.go:452]: Will update the go-spdk-helper.

File: proto/spdkrpc/spdk.proto (hunk 2 above)
Comment (derekbit) [reply proto/spdkrpc/spdk.proto:103]: I'm fine with the default values and not expose them for now.
WDYT?

Comment (shuo-wu) [reply proto/spdkrpc/spdk.proto:103]: I am fine with either. We can expose them when required.

**Comment by derekbit** [comment]
expose them when required in the future.

**Pull request closed without merging**